envoy/config/filter/http/health_check/v2/health_check.proto

```diff
@@ -30,4 +31,11 @@ message HealthCheck {
   // names and the minimum percentage of servers in each of those clusters that
   // must be healthy in order for the filter to return a 200.
   map<string, envoy.type.Percent> cluster_min_healthy_percentages = 4;
+
+  // [#not-implemented-hide:]
+  // Specifies a set of health check request headers to match on. The health check filter will
+  // check a request’s headers against all the specified headers. A request will be considered a
+  // health check request if all headers are present. Notice that if specificed, the ``:path``
```

Comment:
My vote here is to just have `endpoint` implicitly set the `:path` header and internally just do header matching. You could say that if the `endpoint` header is set it will overwrite any set `:path` header. I think it's probably fine to mark `endpoint` deprecated (`[deprecrated]`) and document that setting `:path` should be used instead?

Reply:
Yes sounds good. 

Is the workflow to send out a PR in envoyproxy/envoy to implement this feature, then send out a 2nd PR to add the [deprecated] tag to the endpoint field?

Reply:
I think you can add the deprecated tag now. It doesn't do anything other than inform people. We don't even need to explicitly drop support for endpoint if we don't want.